semanticz-connector-graphdb/src/main/java/zone/cogni/semanticz/connectors/graphdb/GraphDBRdfStoreService.java

```diff
@@ -0,0 +1,215 @@
+                .PUT(HttpRequest.BodyPublishers.ofString(turtle, StandardCharsets.UTF_8))
+                .build();
+
+        executeHttpRequest(request, 200, 202, 204);
+        log.debug("Graph {} successfully replaced with {} triples.", graphUri, model.size());
+    }
+
+    @Override
+    public <R> R executeSelectQuery(Query query,
+                                    QuerySolutionMap bindings,
+                                    JenaResultSetHandler<R> resultHandler,
+                                    String context) {
+
+        QueryExecutionHTTPBuilder builder = QueryExecutionHTTP.service(config.getSparqlEndpoint())
+                .query(query)
+                .httpClient(getHttpClient());
+
+        if (hasBindings(bindings)) {
+            builder.substitution(bindings);
+        }
+
+        if (StringUtils.isNotBlank(context)) {
+            builder.param("context-uri", context);
+        }
+
+        try (QueryExecution qexec = builder.build()) {
+            ResultSet rs = qexec.execSelect();
+            return resultHandler.handle(rs);
+        }
+    }
+
+    @Override
+    public boolean executeAskQuery(Query query, QuerySolutionMap bindings) {
+        QueryExecutionHTTPBuilder builder = QueryExecutionHTTP.service(config.getSparqlEndpoint())
+                .query(query)
+                .httpClient(getHttpClient());
+
+        if (hasBindings(bindings)) {
+            builder.substitution(bindings);
+        }
+
+        try (QueryExecution qexec = builder.build()) {
+            return qexec.execAsk();
+        }
+    }
+
+    @Override
+    public Model executeConstructQuery(Query query, QuerySolutionMap bindings) {
+        QueryExecutionHTTPBuilder builder = QueryExecutionHTTP.service(config.getSparqlEndpoint())
+                .query(query)
+                .httpClient(getHttpClient());
+
+        if (hasBindings(bindings)) {
+            builder.substitution(bindings);
+        }
+
+        try (QueryExecution qexec = builder.build()) {
+            return qexec.execConstruct();
+        }
+    }
+
+    @Override
+    public void executeUpdateQuery(String updateQuery) {
+        String formBody = "update=" + URLEncoder.encode(updateQuery, StandardCharsets.UTF_8);
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(config.getSparqlUpdateEndpoint()))
+                .header("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
+                .POST(HttpRequest.BodyPublishers.ofString(formBody))
+                .build();
+
+        executeHttpRequest(request, 204);
+        log.debug("SPARQL‑UPDATE executed successfully (204 NO CONTENT)");
+    }
+
+    @Override
+    public void delete() {
+        executeUpdateQuery("CLEAR ALL");
+        log.info("Repository {} cleared with 'CLEAR ALL'.", config.getRepository());
+    }
+
+    @Override
+    public void close() {
+        RdfStoreService.super.close();
```

Comment:
weird. super.close() is not possible?

Reply:
super.close(); tries to resolve method close from superclass which is Object